# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmJS.cpp
# Commit: f655a49fb58f
# Full Hash: f655a49fb58f1984ebf5f89019dd27b7fd8530f9
# Author: Ms2ger <Ms2ger@gmail.com>
# Date: 2021-09-07 21:47:56
# Description:
#   Bug 1729266 - Add missing null check in WasmFunctionCreate; r=lth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D124618
# ==============================================================================

diff -r 870c8ca46ce6 -r f655a49fb58f js/src/wasm/WasmJS.cpp
--- a/js/src/wasm/WasmJS.cpp	Tue Sep 07 12:48:23 2021 +0000
+++ b/js/src/wasm/WasmJS.cpp	Tue Sep 07 13:25:29 2021 +0000
@@ -4160,6 +4160,10 @@
   ScriptedCaller scriptedCaller;
   SharedCompileArgs compileArgs =
       CompileArgs::build(cx, std::move(scriptedCaller), options);
+  if (!compileArgs) {
+    return nullptr;
+  }
+
   ModuleEnvironment moduleEnv(compileArgs->features);
   CompilerEnvironment compilerEnv(CompileMode::Once, Tier::Optimized,
                                   OptimizedBackend::Ion, DebugEnabled::False);
